### PR TITLE
CSV Export Adjustments & Breaking API/DB Typos

### LINKED ISSUE #12: hottnes should be renamed to hotness
Appears in Comment and related resolvers and lib methods.

### PR DESCRIPTION
**Pull request by pozylon**
Tasks:
- Adds one new column for every package option: "X wert" (output 240 on reduced abos, else the price payed)
- Duplicates all cancelled payments adding a simulated second line per payment reflecting the "sale" booking
- Adds the following columns to the CSV export: type (verkauf or storno), simulatedBookingDate (createdAt or updatedAt depending on verkauf/storno), total (inverted price depending on verkauf/storno) 
- companyName as mandatory query param for paymentsCSV

Breaking for UI's:
- GraphQL Queries using "Comment" type: Comment.hottnes is now Comment.hotness
- GraphQL Root Query "paymentsCSV": companyName is required
- Goodie name is now "TOTEBAG"

Breaking for Mail templates:
- NOTEBOOK_OR_TOADBAG -> NOTEBOOK_OR_TOTEBAG
This closes #12 .

So this issue needs to get merged in concert with:

https://github.com/orbiting/republik-frontend/pull/39
https://github.com/orbiting/crowdfunding-admin/pull/7

### CODE REVIEW

**Comment by pozylon** [comment]
Yo @patte please review

**Review by patte — CHANGES_REQUESTED** [review]
Looks good to me, nice and clean 👍 
Some things tough:
- the export should take a company name as input and only include payments for this entitity
- the new TOADBAG goodie should be added. Optional: If you want to go the long way it would be very nice if you could rename TOADBAG to TOTEBAG throughout the code (notify clara if it changes in mailchimp templates)
- if you didn't do so already, could you please check throughout the code if payment.updatedAt is only changed upon cancel, not to mess up the verkauf and storno dates.
- only if it's not too hard: it would be nice if you could integrate `modules/crowdfundings/lib/Pledge.js` to get the pledge details. It would be nice to have all this logic in one place, but maybe it's also a bad idea to use it here, you can decide.

**Comment by pozylon** [comment]
Breaking for UI's:
- GraphQL Queries using "Comment" type: Comment.hottnes is now Comment.hotness
- GraphQL Root Query "paymentsCSV": companyName is required
- Goodie name is now "TOTEBAG"

Breaking for Mail templates:
- NOTEBOOK_OR_TOADBAG -> NOTEBOOK_OR_TOTEBAG

This closes #12 .

**Comment by pozylon** [comment]
This closes #12 btw.

**Comment by pozylon** [comment]
> only if it's not too hard: it would be nice if you could integrate modules/crowdfundings/lib/Pledge.js to get the pledge details. It would be nice to have all this logic in one place, but maybe it's also a bad idea to use it here, you can decide.

I think this is out of scope of this pull request, I'd rather sketch out a plan on how to build a more central domain model in general.

**Comment by pozylon** [comment]
Just checked on the updatedAt case:

payment.updatedAt gets updated not only during cancel, it always gets updated with every change of the payment status, but that's fine as we only trust that date in the "storno" case and storno is the last step in the workflow. So from my perspective, dates should be fine. At least they are enough "accurate" (correct month) for financial purposes. What do you think?

File: migrations/sqls/20180103132238-rename-hottnes-toadbag-down.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE "comments" RENAME COLUMN hotness TO hottnes;
+UPDATE "goodies" SET "name"='TOTEBAG' WHERE "name" = 'TOTEBAG';
```
Comment (patte) [inline migrations/sqls/20180103132238-rename-hottnes-toadbag-down.sql]: typo
`UPDATE "goodies" SET "name"='TOADBAG' WHERE "name" = 'TOTEBAG';`

**Comment by patte** [comment]
HAPPY NEW YEAR! 🎉 
Great work so far!
- totally agree with the Pledge.js integration, postponed.
- It's great that you also fixed hottness! For both the renames I would prefer not to do it with a new migration, but to change `migrations/sqls/20170724202357-migrate-crowdfunding-up.sql` for hotness and in the same file add the migration for toad-tote. As long as we (or others) don't have this code in production I prefer to keep the migrations count low.
- could you make the necessary changes / a PR in [crowdfunding-admin](https://github.com/orbiting/crowdfunding-admin) to include the companyName in the download request and show two download links, one for each company? If something is unclear with the crowdfunding-admin repo don't hesitate to call me.

File: migrations/sqls/20180103132238-rename-hottnes-toadbag-down.sql (hunk 1 above)
Comment (patte) [reply migrations/sqls/20180103132238-rename-hottnes-toadbag-down.sql]: obsolete

**Comment by pozylon** [comment]
Thanks @patte. Same to you!
Okay this should be ready now 😀
  
As soon as I have done the PR for crowdfunding-admin, this one here needs to get merged in concert with:

https://github.com/orbiting/republik-frontend/pull/39
https://github.com/orbiting/crowdfunding-admin/pull/7

**Comment by patte** [comment]
@pozylon you could take this branch: https://github.com/orbiting/crowdfunding-admin/pull/7

**Review by patte — APPROVED** [review]

**Comment by clarajeanne** [comment]
Thx for the export. Looks great!
Issues I found:
- A comma in the name field leads to wrong representation. Example: pledge-id afbe7944-2aa8
- The ABO_total row should be the product of ABO_wert and # ABO, same for all Product_total fields
- ABO_wert should always be 240, even if there is no ABO in this pledge, same for all Product_wert fields
- Row ABO_REDUCED total should show the price that was paid, not 240

**Pull request merged**